### PR TITLE
[WIP] Unit testing in v6

### PR DESCRIPTION
**Pull request by hmemcpy**
Documenting changes to the unit testing approach in v6, supporting "bare metal", AAA-style tests without using the .Testing framework.

### CODE REVIEW

**Comment by hmemcpy** [comment]
This is a very early PR to discuss unit-testing docs in v6: what kind of API do we want, etc. Text and code snippets will change.

File: nservicebus/testing/index.md [hunk 1]
```diff
@@ -7,7 +7,9 @@ redirects:
  - nservicebus/testing/unit-testing
 ---
 
-Developing enterprise-scale distributed systems is hard and testing them is just as challenging a task. The architectural approach supported by NServiceBus makes these challenges more manageable. And the testing facilities provided actually make unit testing endpoints and workflows easy. You can now develop service layers and long-running processes using test-driven development.
+NOTE: The approach for testing NServiceBus has been changed significantly in Version 6, to allow for "bare-metal", Arrange-Act-Assert (AAA) style tests. Please refer to [Unit Testing NServiceBus](/nservicebus/testing/unit-testing.md) for more information.
```
Comment (SimonCropp) [inline nservicebus/testing/index.md]: how about 

> See also [Unit Testing NServiceBus](/nservicebus/testing/unit-testing.md).

File: nservicebus/testing/index.md [hunk 2]
```diff
@@ -7,7 +7,9 @@ redirects:
...
+
+Developing enterprise-scale distributed systems is hard and testing them is just as challenging a task. The architectural approach supported by NServiceBus makes these challenges more manageable. The testing facilities provided make unit testing endpoints and workflows easy, allowing developing service layers and long-running processes using Test-Driven Development.
```
Comment (SimonCropp) [inline nservicebus/testing/index.md]: > Developing enterprise-scale distributed systems is hard and testing them is just as challenging a task. The architectural approach supported by NServiceBus makes these challenges more manageable.

remove the marketing speak

Comment (SimonCropp) [inline nservicebus/testing/index.md]: > The testing facilities provided make unit testing endpoints and workflows easy

dont say it is easy. that should be clear from the description/snippets that illustrate it

File: nservicebus/testing/index.md [hunk 3]
```diff
@@ -66,4 +70,6 @@ snippet:TestingAdditionalDependencies
 
 ## Constructor injected bus
 
+NOTE: In Version 6, the `IBus` interface was deprecated and removed, and replaced with the contextual `IMessageHandlerContext` parameter on the `IHandleMessages<T>.Handle()` methods.
```
Comment (SimonCropp) [inline nservicebus/testing/index.md]: Versions 6 and above

File: nservicebus/testing/unit-testing.md [hunk 4]
```diff
@@ -0,0 +1,53 @@
+---
+title: Unit Testing NServiceBus 6  
+summary: Write unit tests for NServiceBus 6 components
+tags: []
+redirects:
+---
+
+In Version 6 of NServiceBus we've redesigned core parts of the system, allowing more granular support for testing various components, such as Handlers, Sagas and even Behaviors, without needing to spin up an actual endpoint for end-to-end test scenarios. This allows you to write very fast "bare-metal" unit tests for those components, utilizing convenience methods and test-doubles.
```
Comment (SimonCropp) [inline nservicebus/testing/unit-testing.md]: versions 6 and above

File: nservicebus/testing/unit-testing.md [hunk 5]
```diff
@@ -0,0 +1,53 @@
+---
+title: Unit Testing NServiceBus 6
```
Comment (SimonCropp) [inline nservicebus/testing/unit-testing.md]: version 6 and above

File: nservicebus/testing/unit-testing.md [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+summary: Write unit tests for NServiceBus 6 components
+tags: []
+redirects:
+---
+
+In Version 6 of NServiceBus we've redesigned core parts of the system, allowing more granular support for testing various components, such as Handlers, Sagas and even Behaviors, without needing to spin up an actual endpoint for end-to-end test scenarios. This allows you to write very fast "bare-metal" unit tests for those components, utilizing convenience methods and test-doubles.
+
+## Getting started
+
+To get started, create a test project using your favorite framework, like NUnit, XUnit, or MSTest. Add a reference to the NServiceBus 6 NuGet package, and we're ready to go!
```
Comment (SimonCropp) [inline nservicebus/testing/unit-testing.md]: > and we're ready to go!

remove it

File: nservicebus/testing/unit-testing.md [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+
+## Unit testing a Handler
+
+Here's an example of testing a `Reply`:
+
+```csharp
+[Test]
+public async Task Should_Reply()
```
Comment (SimonCropp) [inline nservicebus/testing/unit-testing.md]: put this in a snippet

File: nservicebus/testing/unit-testing.md [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+  context.Sent.ShouldContain(m => m....);
+  context.Published.ShouldBeEmpty();
+}
+```
+
+## Unit testing a Saga
+
+```csharp
+[Test]
+public async Task Saga_Should_Store_Data()
```
Comment (SimonCropp) [inline nservicebus/testing/unit-testing.md]: put this in a snippet

File: nservicebus/testing/index.md (hunk 3 above)
Comment (SimonCropp) [inline nservicebus/testing/index.md]: > was deprecated and removed

to 

> is deprecated

File: nservicebus/testing/unit-testing.md (hunk 4 above)
Comment (SimonCropp) [inline nservicebus/testing/unit-testing.md]: remove "you"

File: nservicebus/testing/unit-testing.md (hunk 6 above)
Comment (SimonCropp) [inline nservicebus/testing/unit-testing.md]: remove your

File: nservicebus/testing/unit-testing.md (hunk 4 above)
Comment (SimonCropp) [inline nservicebus/testing/unit-testing.md]: remove "we've"

**Comment by SimonCropp** [comment]
@hmemcpy how is this related to https://github.com/Particular/docs.particular.net/pull/1305

**Comment by hmemcpy** [comment]
@simon I wanted it to be a continuation, a spin-off that PR. But after some consideration, I'm going to make this into a complete sample instead.
I'll close this PR and send another one.

**Comment by SimonCropp** [comment]
ok cool. sorry for the hassle

**Comment by hmemcpy** [comment]
No worries! I'll send it after #1305 is merged, want to base it on top of it.

**Pull request closed without merging**